.huskyrc.js

```diff
@@ -15,23 +15,31 @@ const svg2Components = {
 
 // lintStaged actually just runs formatting rules on staged files
 const lintStaged = path => {
-  return {command: 'lint-staged', path: path} 
+  return { command: "lint-staged", path: path }
 }
 
-// Run eslint on everything (used pre-push)
+// Run eslint on the files inside path for the last commit
+// It will try to amend the latest commit if possible to fix.
 const eslint = path => {
-  return { command: 'eslint --fix .', path: path }
+  return {
+    command: `git diff-tree --no-commit-id --name-only -r HEAD . | sed 's/${path}\\///' | xargs eslint --fix`,
```

Comment:
Awk and Sed are the 2 most important tools!

LG!

Reply:
More than important: the only ones you need. I think awk and sed are turing complete!